pkg/registry/core/node/rest/proxy.go

```diff
@@ -70,7 +70,9 @@ func (r *ProxyREST) Connect(ctx genericapirequest.Context, id string, opts runti
 	if err != nil {
 		return nil, err
 	}
-	location.Path = path.Join("/", location.Path, proxyOpts.Path)
+	if location.Path != "" || proxyOpts.Path != "" {
```

Comment:
Actually, I’m not sure this is right… what gets populated in proxyOpts.Path for “.../nodes/foo/proxy” and “.../nodes/foo/proxy/“?

Reply:
Empty strings I suppose. The if condition only affects the case when both location.Path and proxyOpts.Path are empty.

Reply:
I think the right fix is to fix the way proxyOpts.Path is populated for proxy subresources to preserve any leading and trailing slashes. Opened https://github.com/kubernetes/kubernetes/pull/52933 to do that instead